### PR TITLE
Add necessary commits for 2.4.3

### PR DESCRIPTION
**Pull request by IAlibay**
Cherry picked commits for 2.4.3:
* Fixes DCD seeking for large (2Gb+) files.

Commits to prepare for 2.4.3:
* Update version numbers

### CODE REVIEW

**Comment by IAlibay** [comment]
cirrus failure is expected - I don't see a need to cherry pick the fix here (since we don't use it for release), unless folks think otherwise

**Comment by IAlibay** [comment]
I guess we'll need to cherry pick the parmed fix here.

**Comment by IAlibay** [comment]
:/ this is a bit of a difficult one, if we cherry-pick the parmed fix, do we then change the changelog on develop to reflect that it went in on 2.4.3? Or do we just do it here and leave it as-is?

The alternative option is to pin the upper value of Parmed in tests & conda-forge?

**Comment by IAlibay** [comment]
Note - it's a fix on a failing test, but it didn't require any change to the Parmed parser codes.

**Comment by IAlibay** [comment]
Pinging @MDAnalysis/coredevs it'd be good to get a quick decision on the path forward here.

**Comment by richardjgowers** [comment]
if the parmed fix is possible via a conda-forge pin it might make sense to do 2.4.3 that way?

File: package/CHANGELOG [hunk 1]
```diff
@@ -13,6 +13,15 @@ The rules for this file:
...
 
 ------------------------------------------------------------------------------
+03/29/23 richardjgowers
+
+ * 2.4.3
+
+Fixes
+  * Fixed DCD reading for large (>2Gb) files (Issue #4039).  This was broken
+    for versions 2.4.0, 2.4.1 and 2.4.2
+
```
Comment (orbeckst) [inline package/CHANGELOG]: Just as a reminder. We have to retroactively change the main develop CHANGELOG to include 2.4.3 and reshuffle entries.

**Review by orbeckst — APPROVED** [review]
If the parmed fix thing does not require an increase in minor version then you can put it in here without problems (especially if it's just something in the tests). 

As with anything else, we just need to retroactively change develop's CHANGELOG.

EDIT: and add yourself to the 2.4.3 authors list @IAlibay

File: package/CHANGELOG (hunk 1 above)
Comment (orbeckst) [reply package/CHANGELOG]: I just saw that you had already snuck it into PR #4048 ...

**Comment by IAlibay** [comment]
Ok, given we need to update the author list anyways, I'll just cherry-pick the parmed test fix and separately change the changelog for `develop` (these bugfix changes don't guarantee a oorresponding history with `develop` anyways, so it being a bit out of synch doesn't matter).

**Comment by RMeli** [comment]
Spack updated: https://github.com/spack/spack/pull/36541

**Pull request merged**